neard/src/main.rs

```diff
@@ -180,6 +181,13 @@ fn main() {
             );
         }
         ("run", Some(args)) => {
+            // check disk
+            let system = sysinfo::System::new_all();
+            for disk in system.get_disks() {
+                if !matches!(disk.get_type(), sysinfo::DiskType::SSD) {
+                    warn!("!!! DETECTED NON-SSD DISK. PLEASE MAKE SURE THAT YOU USE SSD FOR NEAR NODE !!!");
+                }
+            }
```

Comment:
We should only care about the storage type behind the home dir. Here you require the machine has SSD drives installed into it physically (right?) while the home dir might be mounted via network and this check will be useless. I think we should not make such requirements and especially require ALL the drives on the machine to be SSD. We should monitor the runtime performance and only print such scary warnings if we observe that the IO performance is the bottleneck.

Reply:
> We should only care about the storage type behind the home dir.

Yes, but I didn't find a good way to do that. Do you have any suggestions?

> We should monitor the runtime performance and only print such scary warnings if we observe that the IO performance is the bottleneck.

We did and concluded that HDD just cannot work.

Reply:
Not sure if it works on a mac, but:

Execute `df` on the data folder, it will give you among other things the disk it is on.
Then execute `lsblk -d -o name,rota` and grep for that drive.

@frol a node with a HDD will not be able to keep up with the network, so it makes sense to not even allow starting it, so that people don't waste time on syncing (which on HDD will take a lot of time). Detecting slowness real time will be more flaky than to detect that the data folder is on an HDD.

Reply:
Do we use random access so wildly? (Just want to learn how a database of a few GB can already be a problem for HDD with a reasonable buffet size)

Reply:
@frol right now the biggest problem is GC, it sometimes takes 2 seconds to delete data.

But generally every transaction is random access, because they read from random places on disk.